src/goto-symex/goto_symex.cpp

```diff
@@ -27,6 +30,21 @@ void goto_symext::replace_nondet(exprt &expr)
   if(expr.id()==ID_side_effect &&
      expr.get(ID_statement)==ID_nondet)
   {
+    null_message_handlert null_message;
+    exprt nondet=
+      nondet_initializer(
+        expr.type(),
+        expr.source_location(),
+        ns,
+        null_message);
+
+    if(nondet.id()!=ID_side_effect)
```

Comment:
Comment (in the code) what `nondet` could be? Under what circumstance does it get replaced or not replaced?